### PR TITLE
Restructure project files idiomatically, fix errors and clippy warnings, updated deps

### PR DESCRIPTION
**Pull request by piedoom**
Hello! Sorry for the big PR, obviously feel free to reject if this if it doesn't align where you want to take the project.

I was surprised to find that this is just about the only `embedded-rust` GUI project outside of slint (which I find a little too heavy for my liking). I began writing my own small (and bad) library before I found this via search, and want to try and adopt and contribute back. This PR accomplishes the following without any breaking API changes:

1. Structure examples idiomatically (can now be run like `cargo run --example basic-example`
    a. This also is beneficial in that RA detects these files
    b. I fixed the `motion-scheduler` example (too many generic params) after making this change
3. Removed some dead/temporary code
4. Updated all dependencies and updated to 2024 edition
5. Resolved all warnings, except for 1: `SmartstateProvider::next` warns that it may be confused with the `Iterator` next. I didn't change anything in this PR, but I might recommend it be renamed to something like `step` or `tick`.

Note that I do not use IntelliJ, so the `.idea` project files might be invalid.  

Thanks!

### CODE REVIEW

**Comment by Yandrik** [comment]
Yep :D The significant lack of other frameworks was the exact reason I started this, but I never brought it to a proper release sadly. Thanks for the PR!
I really like what you did with it. Only thing that I'd like to keep for now is the 2021 edition, until 2024 is officially stabilized. Once 2024 is officially out, switching it over is definitely a good idea. Other than that I like the restructure. I'm not quite sure why I had the other structure before, but this is honestly significantly more convenient. 

With this and the integration of what is at the moment the `optimizations` branch (which contains speed improvements, but also Sliders and Toggles, as well as various fixes), I believe this is almost ready for a 0.1.0 release (especially considering that it's currently still alpha)!

**Comment by Yandrik** [comment]
I already integrated your changes into the `optimizations-restructure` branch, so merging it on top of the MR once it's done should be pretty easy :)

File: Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+[package]
+name = "kolibri-embedded-gui"
+version = "0.0.0-alpha.1"
+edition = "2024"
```
Comment (Yandrik) [inline Cargo.toml:4]:
```suggestion
edition = "2021"
```

If you apply this I'd consider this ready to merge :)

**Comment by piedoom** [comment]
ack, committed other stuff on the wrong branch, sorry about that, ill try and fix

**Pull request merged**